internal/connector/connector.go

```diff
@@ -93,6 +115,7 @@ func Run(ctx context.Context, options Options) error {
 	}
 
 	// server is localhost which should never be the case. try to infer the actual host
+	// TODO: do this lookup when Host is "" instead of localhost
```

Comment:
I created #3380 for this issue, I'll add the link to this comment in a follow up.